crates/steel-parser/src/parser.rs

```diff
@@ -478,9 +478,26 @@ impl<'a> Parser<'a> {
         }
     }
 
-    fn read_from_tokens(&mut self) -> Result<ExprKind> {
-        let mut stack: Vec<Vec<ExprKind>> = Vec::new();
-        let mut current_frame: Vec<ExprKind> = Vec::new();
+    fn maybe_lower_frame(&self, frame: Frame, close: Span) -> Result<ExprKind> {
```

Comment:
We can _almost_ get rid of the regular `maybe_lower`, if we short-circuit the only remaining call when parsing quotes ([here](https://github.com/mattwparas/steel/blob/842dd029978f77298768bd8f8913e3a4ef242a75/crates/steel-parser/src/parser.rs#L1022)) to generate a `ExprKind::Quote` directly.